dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,682 @@
+import com.dotmarketing.business.web.HostWebAPI;
+import com.dotmarketing.business.web.UserWebAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.RegEX;
+import com.dotmarketing.util.SecurityLogger;
+import com.dotmarketing.util.UtilMethods;
+import com.dotmarketing.util.WebKeys;
+import com.dotmarketing.util.json.JSONException;
+import com.liferay.portal.PortalException;
+import com.liferay.portal.SystemException;
+import com.liferay.portal.auth.PrincipalThreadLocal;
+import com.liferay.portal.model.User;
+import com.liferay.portal.servlet.PortletSessionPool;
+import io.vavr.control.Try;
+import org.apache.commons.lang.StringUtils;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.dotcms.saml.DotSamlConstants.SAML_USER_ID;
+
+/**
+ * This interceptor encapsulates the logic for Saml
+ * Basically if there is any configuration set on apps and c
+ * @author jsanca
+ */
+public class SamlWebInterceptor implements WebInterceptor {
+
+    public static final String APPS_SAML_CONFIG_KEY   = "app-saml-config";
+    public static final String REFERRER_PARAMETER_KEY = "referrer";
+    public static final String ORIGINAL_REQUEST       = "original_request";
+    public static final String BY_PASS_KEY            = "native";
+    public static final String BY_PASS_VALUE          = "true";
+
+    protected final Encryptor       encryptor;
+    protected final LoginServiceAPI loginService;
+    protected final UserAPI         userAPI;
+    protected final UserWebAPI      userWebAPI;
+    protected final HostWebAPI      hostWebAPI;
+    protected final AppsAPI         appsAPI;
+    protected final IdentityProviderConfigurationFactory identityProviderConfigurationFactory;
+    protected final SamlConfigurationService             samlConfigurationService;
+    protected final SamlAuthenticationService            samlAuthenticationService;
+
+    public SamlWebInterceptor() {
+        this(APILocator.getLoginServiceAPI());
+    }
+
+    public SamlWebInterceptor(final LoginServiceAPI loginServiceAPI) {
+        this.loginServiceAPI = loginServiceAPI;
+    }
+
+    @Override
+    public Result intercept(final HttpServletRequest request,
+                            final HttpServletResponse response) throws IOException {
+
+        HttpSession session = request.getSession(false);
+
+        if (this.isByPass(request, session)) {
+
+            Logger.info(this, ()->"Using SAML by pass");
+            return Result.NEXT;
+        }
+
+        Result result      = Result.NEXT;
```

Comment:
Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)